django/db/models/lookups.py

```diff
@@ -431,7 +431,10 @@ def get_group_by_cols(self, alias=None):
             if not getattr(self.rhs, 'has_select_fields', True):
                 self.rhs.clear_select_clause()
                 self.rhs.add_fields(['pk'])
-            cols.extend(self.rhs.get_group_by_cols())
+            from django.db.models.sql.query import \
+                Query  # avoid circular import
+            if not isinstance(self.rhs, Query):
+                cols.extend(self.rhs.get_group_by_cols())
```

Comment:
This unfortunately won't work if the subquery refers to any outer references as these must be included in the group by clause.